src/App.js

```diff
@@ -1,16 +1,24 @@
 import React from "react";
 import { toast } from "react-toastify";
 import "react-toastify/dist/ReactToastify.css";
+import { ThemeProvider as ThemeProviderMaterial } from "@material-ui/core/styles";
+import { ThemeProvider as ThemeProviderStyledCmp } from "styled-components";
 
+import { theme, themeMaterial } from "assets/theme";
+import * as Styles from "./styles";
 import { MainPage } from "pages";
 
 toast.configure();
 
 function App() {
   return (
-    <div className="App">
-      <MainPage />
-    </div>
+    <ThemeProviderStyledCmp theme={theme}>
```

Comment:
Są dwie szkoły używania swoich themów w styled components. Ja np. nie lubię używać tego przez propsy, wolę importować bezpośrednio z pliku obiekt, tj.
export const color = {
...
}

i potem w styles.ts

Cośtam = styled.div`
color: ${color.grey}
`

Przy importach jest trochę mniej kodu, niż pisać wszędzie props => props.theme.color, tylko po to, by wyciagnac kolor. Za to theme bardzo sie przydaje, kiedy trzeba dorobic drugi i je miedzy soba przelaczac

Reply:
To taki komentarz po prostu, ze mozesz tez sprobowac w ten drugi sposob i zobaczyc czy Ci sie bardziej podoba

Reply:
Rozumiem dzięki za taką wiedzę bo mam porównanie różnych sposobów a to zawsze otwiera szerzej obraz tego co już znam :) Wybiorę tą opcję którą zastosowałem z tą różnicą że do VSC dodałem sobie User Snippets. 
"Import styled Components Theme": {
    "prefix": "isct",
    "body": ["$1: ${(props) => props.theme.$2};", "$0"]
  },

:) ale super że pokazujesz mi alternatywy możesz zawsze mi takie wysyłać jak wpadną Ci do głowy xd

Reply:
Done.

Reply:
Zmieniam zdanie xd Rozbudowałem theme tak aby w nim znajdowały się media colors i css-y i teraz to ma sens jeszcze jak się zrobi destrukturyzację to już wgl wypas :).
Biere to!
Done.